lib/archethic/p2p/client/connection.ex

```diff
@@ -37,14 +40,22 @@ defmodule Archethic.P2P.Client.Connection do
           | {:error, :closed}
   def send_message(public_key, message, timeout \\ 3_000) do
     ref = make_ref()
-    GenStateMachine.cast(via_tuple(public_key), {:send_message, ref, self(), message, timeout})
-
-    receive do
-      {^ref, msg} ->
-        msg
-    after
-      timeout ->
-        {:error, :timeout}
+
+    case GenStateMachine.call(
+           via_tuple(public_key),
```

Comment:
viaTuple?

Reply:
This is used to redirect to a specific pid for a given public key